### PR TITLE
Default LogsDB value for `ignore_dynamic_beyond_limit`

### PR DESCRIPTION
**Pull request by salvatore-campagna**
This is a backport of #115265

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/index/IndexVersions.java [hunk 1]
```diff
@@ -118,6 +118,8 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion MERGE_ON_RECOVERY_VERSION = def(8_515_00_0, Version.LUCENE_9_11_1);
     public static final IndexVersion UPGRADE_TO_LUCENE_9_12 = def(8_516_00_0, Version.LUCENE_9_12_0);
     public static final IndexVersion ENABLE_IGNORE_ABOVE_LOGSDB = def(8_517_00_0, Version.LUCENE_9_12_0);
+    public static final IndexVersion ADD_ROLE_MAPPING_CLEANUP_MIGRATION = def(8_518_00_0, Version.LUCENE_9_12_0);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/index/IndexVersions.java:121]: This version should already be present on 8.x

File: server/src/main/java/org/elasticsearch/index/IndexVersions.java [hunk 2]
```diff
@@ -118,6 +118,8 @@ private static IndexVersion def(int id, Version luceneVersion) {
...
+    public static final IndexVersion LOGSDB_DEFAULT_IGNORE_DYNAMIC_BEYOND_LIMIT = def(8_519_00_0, Version.LUCENE_9_12_0);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/index/IndexVersions.java:122]: For consistency, this should be named the same as in main, to highlight that its a backport

**Review by thecoop — APPROVED** [review]
IndexVersion LGTM, but this should be updated from latest 8.x to ensure this PR is only adding the index version related to this change

File: server/src/main/java/org/elasticsearch/index/IndexVersions.java (hunk 2 above)
Comment (salvatore-campagna) [reply server/src/main/java/org/elasticsearch/index/IndexVersions.java:122]: Yes I will update the backport PR to have the name matching. Thanks.

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Pull request merged**